user-interface/src/main/java/life/qbic/datamanager/views/projects/purchase/UploadPurchaseDialog.java

```diff
@@ -0,0 +1,193 @@
+
+  private static final int MAX_FILE_SIZE_BYTES = 1024 * 1024 * 14; // 14 MB
+
+  @Serial
+  private static final long serialVersionUID = 6602134795666762831L;
+  private final Upload upload;
+  private EditableMultiFileMemoryBuffer multiFileMemoryBuffer;
+  private Div uploadedPurchaseItems;
+
+  private List<PurchaseItem> purchaseItemsCache = new ArrayList<>();
+
+  private Div disclaimer;
+
+  private Div titleBox;
+
+  private List<ComponentEventListener<ConfirmEvent<UploadPurchaseDialog>>> confirmListeners = new ArrayList<>();
+
+  public UploadPurchaseDialog() {
+    // Vaadin's upload component setup
+    multiFileMemoryBuffer = new EditableMultiFileMemoryBuffer();
+    upload = new Upload(multiFileMemoryBuffer);
+    upload.setAcceptedFileTypes("application/pdf", ".pdf");
+    upload.setMaxFileSize(MAX_FILE_SIZE_BYTES);
+
+    // Title box configuration
+    titleBox = new Div();
+    var title = new Span("Upload your offer");
+    title.addClassName("title");
+    titleBox.add(title);
+    titleBox.addClassName("title-box");
+
+    // Upload restriction display configuration
+    var restrictions = new Div();
+    restrictions.addClassName("restrictions");
+    restrictions.add(new Span("Accepted file formats: PDF (.pdf)"));
+    restrictions.add(
+        new Span("Maximum file size: %s MB".formatted(MAX_FILE_SIZE_BYTES / (1024 * 1024))));
+
+    // Uploaded purchase items display configuration
+    uploadedPurchaseItems = new Div();
+    uploadedPurchaseItems.addClassName("uploaded-documents");
+    disclaimer = new Div();
+    disclaimer.addClassName("instructions");
+    disclaimer.add(new Span("Set offer signed status"));
+    disclaimer.add(new Paragraph("Please tick the checkbox if the offer is signed."));
+
+
+    // Add upload offers to the purchase item section, where users can set the signed flag
+    upload.addSucceededListener(succeededEvent -> {
+      var purchase = new PurchaseItem(succeededEvent.getFileName());
+      uploadedPurchaseItems.add(purchase);
+      purchaseItemsCache.add(purchase);
+      toggleFileSectionIfEmpty();
+    });
+
+    // Synchronise the Vaadin upload component with the purchase list display
+    // When a file is removed  from the upload component, we also want to remove it properly from memory
+    // and from any additional display
+    upload.getElement().addEventListener("file-remove", this::processClientFileRemoveEvent)
+        .addEventData(VAADIN_FILENAME_EVENT);
+
+    // Put the elements together
+    add(titleBox, upload, restrictions, disclaimer, uploadedPurchaseItems);
+    addClassName("purchase-item-upload");
+    confirmButton.setText("Save");
+
+    // Init the visibility rendering once
+    toggleFileSectionIfEmpty();
+  }
+
+  @Override
+  protected void onConfirmClicked(ClickEvent<Button> clickEvent) {
+    fireConfirmEvent();
+  }
+
+  @Override
+  protected void onCancelClicked(ClickEvent<Button> clickEvent) {
+    this.close();
```

Comment:
should be handled by the calling component in my opinion. This is how it is done with other dialogs.

Reply:
I strongly object, since the abstract parent class method needs to be overridden. An empty code block implementation is bad practise. Either we remove this method from the abstract class to shift the responsibility to the client or we implement it. 
However I would argue that a cancel event is so trivial and the expected behaviour is that the dialog is closed.